tests/js/server/aql/aql-vpack-index-rearming.js

```diff
@@ -0,0 +1,205 @@
+/*jshint globalstrict:true, strict:true, esnext: true */
+/*global AQL_EXPLAIN, assertEqual */
+
+"use strict";
+
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jan Steemann
+////////////////////////////////////////////////////////////////////////////////
+
+const db = require('@arangodb').db;
+const jsunity = require("jsunity");
+const { deriveTestSuite } = require('@arangodb/test-helper');
+
+function VPackIndexRearmingSuite (unique) {
```

Comment:
This test suite is good.
However I think It will also pass if the rearm optimization for some reason is not applied.
Can we add an additional quick test with a failure point inside the Rearm() implementation, such that we run the query (same as in existing tests) and expect the failure point to throw an error. Just to make sure no-one accidentally breaks the rearm logic in future versions.

Reply:
Added a way to track the number of cursor creations and rearmings via tests.